packages/govern-console/src/containers/DAOSettings/DAOSettings.tsx

```diff
@@ -164,14 +164,27 @@ const DaoSettings: React.FC<DaoSettingFormProps> = () => {
     _load();
   }, [daoDetails, provider, config, setValue]);
 
+  const getRule = async (isRuleFile: number | boolean, textRule: string) => {
+    if (Number(isRuleFile) === 1) {
+      if (getValues('rulesFile') instanceof FileList) {
+        // console.log('there is file change', getValues('rulesFile'));
+        return await addToIpfs(getValues('rulesFile')[0]);
+      }
+    } else {
+      if (textRule !== ipfsMetadata?.text) {
+        // console.log('there is text change', textRule);
+        return await addToIpfs(textRule);
+      }
+    }
+    // console.log('no change', config.rules);
+    return config.rules;
+  };
```

Comment:
I haven't looked in detail how the IPFSInput field stuff work yet, but, just a suggestion, may be we can use react-hook-form's formState.isDirty to detect if there's changes to the ipfs value?  This is just a suggestion, I don't know if it works..